File: love.js
const fn = "love.wasm";
const wasmbin = new Uint8Array(await Deno.readFile(fn));
const importObject = {
  env: {
    putchar: (n) => {
      console.log("putchar:", n, String.fromCharCode(n));
    },
  }
};
const wasm = await WebAssembly.instantiate(wasmbin, importObject);
const love = wasm.instance.exports.love;
const love2 = wasm.instance.exports.love2;

console.log(love(Deno.args[0]));
console.log(love2(Deno.args[0]));

File: love.c
extern void putchar(int n);

int love(int param) {
  putchar(65);
  putchar(66);
  putchar(67);
  return 0;
}

int love2(int param) {
  for (int i = 'A'; i <= 'Z'; i++) {
    putchar(i);
  }
  return param * 2;
}
